src/platform_impl/macos/window.rs

```diff
@@ -166,7 +166,17 @@ fn create_window(
                     }
                     None => (800.0, 600.0),
                 };
-                NSRect::new(NSPoint::new(0.0, 0.0), NSSize::new(width, height))
+                let (left, bottom) = match attrs.outer_position {
+                    Some(position) => {
+                        let logical = window_position(position.to_logical(scale_factor));
+                        // macOS wants the position of the bottom left corner,
+                        // but caller is setting the position of top left corner
+                        (logical.x, logical.y - height)
+                    }
+                    // This value is ignored by calling win.center() below
+                    None => (0.0, 0.0),
+                };
```

Comment:
This is unlikely to be solvable. Creating window and then setting its position break the whole point of this feature, which is to avoid the flickering when creating a window and then repositioning it.

I'm also not sure which position is wrong one, whether this one or the one in set_outer_position.

Reply:
Guessing or hardcoding the title bar height is unlikely to work, due to macOS UI changes like this:

https://www.reddit.com/r/MacOSBeta/comments/hq00ll/insane_titlebar_height_and_question_on_logical/

I think we'll have to settle on documenting that this method sets "inner" position on MacOS, despite its name, due to platform limitations.

Reply:
Yep we should definitely not hardcode or guess the menubar height when the function name promises to set the outer position. For the same reason I don't think we should just set the inner position when the function is called `with_outer_position`. But I agree that it would be nice to have at least some way to set the startup position. So at this point the only thing I have problem with is the function name. I think calling it `with_approximate_position` or something similar would be completely fine. And the documentation should of course describe what this means for each platform.

Reply:
What do you think @msiglreith @murarth?